### ISSUE TITLE
@JsonIgnore'd field is included in Model output

### ISSUE THREAD

**Issue by tlovett-rmn** [swagger-api/swagger-core#767]
Here are my dependencies:
```xml
<dependency>
    <groupId>com.wordnik</groupId>
    <artifactId>swagger-annotations</artifactId>
    <version>1.3.10</version>
</dependency>
<dependency>
    <groupId>com.wordnik</groupId>
    <artifactId>swagger-jersey2-jaxrs_2.11</artifactId>
    <version>1.3.10</version>
</dependency>
<dependency>
    <groupId>com.fasterxml.jackson.core</groupId>
    <artifactId>jackson-annotations</artifactId>
    <version>2.4.3</version>
</dependency>
```

Here is my annotated model:
```java
import com.fasterxml.jackson.annotation.JsonIgnore;

@ApiModel(value = "Foo")
public class Foo extends Bar {

    public Foo(Baz baz) {
        super(baz);
    }

    @Override
    @JsonIgnore
    public BazField getBazField() {
        return super.getBazField();
    }
}
```

And despite the fact that BazField is @JsonIgnore'd, and doesn't show up in my API output, it shows up in the swagger api-docs output:
```json
{
  "models": {
    "Foo": {
      "properties": {
        "bazField": {
          "$ref": "bazField"
        }
      }
    }
  }
}
```

Hopefully my obfuscations are correct and not too confusing...

**Comment by todd-richmond** [comment]
Just hit this today. I have a base class "Resource" with a single getUniqueId() method default implementation, but other classes override it (in this case for different Hibernate Id types). I had tried JsonIgnore and hidden=true as well w/o luck

**Comment by fehguy** [comment]
OK this one is tricky, I've traced it down to the `parseRecursive` method inside `ModelPropertyParser.scala`.  It seems that the subclass field is ignored, but added when parsing properties for the superclass.  Looking for a solution...

**Comment by fehguy** [comment]
OK found it--one line fix of course :)  Please try out the 1.3.13-SNAPSHOT which is now on sonatype, or build locally from the develop_scala-2.10 branch.

**Comment by webron** [comment]
**Applies to https://github.com/swagger-api/swagger-core/milestones/v1.5.0-M1 as well.**
